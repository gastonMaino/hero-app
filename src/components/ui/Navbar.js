import React, { useContext } from 'react';
import { Link, NavLink, useHistory } from 'react-router-dom';
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

    const history = useHistory()

    const { user:{name}, dispatch } = useContext(AuthContext);

    const handleLogout = () =>{
        dispatch({
            type: types.logout
        })

        history.replace('/login')
    }

    return (
        <header className='header'>
            <nav className='nav'>

                <Link to='/' className='main-menu__brand'>
                    Asociaciones
                </Link>

                <ul className='main-menu'>
                    <li className='main-menu__item'>
                        <NavLink activeClassName='active' exact to='/marvel' className='main-menu__link'>
                            Marvel
                        </NavLink>
                    </li>
                    <li className='main-menu__item'>
                        <NavLink activeClassName='active' exact to='/dc' className='main-menu__link'>
                            DC
                        </NavLink>
                    </li>
                    <li className='main-menu__item'>
                        <NavLink activeClassName='active' exact to='/search' className='main-menu__link'>
                            Search
                        </NavLink>
                    </li>
                    <li className='main-menu__item'>
                        <button className='main-menu__btn' onClick={handleLogout}>
                            Logout
                        </button>
                    </li>
                </ul>
                <p className='main-menu__user'>
                    {`${name}`}
                </p>
            </nav>
        </header>
    )
}
