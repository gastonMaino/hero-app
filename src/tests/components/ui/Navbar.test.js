import { mount } from "enzyme"
import { MemoryRouter, Router } from "react-router"
import { AuthContext } from "../../../auth/AuthContext"
import { Navbar } from "../../../components/ui/Navbar"
import { types } from "../../../types/types"

describe('tests in <Navbar />', () => {

    const historyMock = {
        replace: jest.fn(),
        push: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'alfonso'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={contexValue}>
            <MemoryRouter>
                <Router history={historyMock}>
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    )

    afterEach(()=>{
        jest.clearAllMocks();
    })

    test('should match to snapshot', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.main-menu__user').text().trim()).toBe('alfonso');
    })

    test('should call logout and use history', () => {
        wrapper.find('button').simulate('click');

        expect(contexValue.dispatch).toHaveBeenCalledWith({ type: types.logout });
        expect(historyMock.replace).toHaveBeenCalledWith('/login');
    })


})
