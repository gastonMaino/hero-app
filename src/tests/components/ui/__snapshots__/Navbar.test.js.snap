// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <Navbar /> should match to snapshot 1`] = `
<header
  className="header"
>
  <nav
    className="nav"
  >
    <a
      className="main-menu__brand"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <ul
      className="main-menu"
    >
      <li
        className="main-menu__item"
      >
        <a
          aria-current={null}
          className="main-menu__link"
          onClick={[Function]}
        >
          Marvel
        </a>
      </li>
      <li
        className="main-menu__item"
      >
        <a
          aria-current={null}
          className="main-menu__link"
          onClick={[Function]}
        >
          DC
        </a>
      </li>
      <li
        className="main-menu__item"
      >
        <a
          aria-current={null}
          className="main-menu__link"
          onClick={[Function]}
        >
          Search
        </a>
      </li>
      <li
        className="main-menu__item"
      >
        <button
          className="main-menu__btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </li>
    </ul>
    <p
      className="main-menu__user"
    >
      alfonso
    </p>
  </nav>
</header>
`;
